File: src/Learn/Arrays101/validMountainArray.cpp
class Solution {
public:
    bool validMountainArray(vector<int>& A) {
        int n = A.size();
        if(n < 3 || A[1] < A[0]) return false;

        bool foundPeak = false;
        int i = 0;
        while(!foundPeak && i < n - 1){
        	if(A[i] < A[i + 1]) i++;
        	else if(A[i] == A[i + 1]) return false;
        	else{
        		foundPeak = true;
        		break;
        	}
        }

        while(foundPeak && i < n - 1){
        	if(A[i] <= A[i + 1]) return false;
        	else i++;
        }

        return foundPeak;
    }
};

File: src/Learn/Arrays101/ValidMountainArray.cs
public class Solution {
    public bool ValidMountainArray(int[] A) {
        int N = A.Length;
        int i = 0;
        // walk up
        while(i + 1 < N && A[i] < A[i + 1]) i++;

        // edge case: peak cannot be first or last element
        if(i == 0 || i == N - 1) return false;

        // walk down
        while(i + 1 < N && A[i] > A[i + 1]) i++;

        return i == N - 1;
    }
}